workflow/controller/operator_test.go

```diff
@@ -2106,3 +2106,109 @@ func findNodeByName(nodes map[string]wfv1.NodeStatus, name string) *wfv1.NodeSta
 	}
 	return nil
 }
+
+var invalidSpec = `
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  name: invalid-spec
+spec:
+  entrypoint: 123
+`
+
+func TestEventInvalidSpec(t *testing.T) {
```

Comment:
good work, again - might be good to add a comment to explain what the test is for

Reply:
done